libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1102,6 +1102,7 @@ bool ExpressionCompiler::visit(FunctionCall const& _functionCall)
 			break;
 		}
 		case FunctionType::Kind::BytesConcat:
+		case FunctionType::Kind::StringConcat:
```

Comment:
Here you will need to case-distinguish in the ``for (auto const& argument: arguments)`` loop below.
If it's ``BytesConcat`` you keep doing what it does now - if it's ``StringConcat``, you only do the equivalent of the ``else`` case, i.e. something like
```
					solAssert(argument->annotation().type->isImplicitlyConvertibleTo(*TypeProvider::stringMemory()), "");
					targetTypes.emplace_back(TypeProvider::stringMemory());
```
I expect the actual implementation after the loop will work out-of-the-box - we can confirm that in semantics tests.